packages/compass-indexes/src/components/create-index-modal/create-index-modal.tsx

```diff
@@ -0,0 +1,252 @@
+import { toggleUsePartialFilterExpression } from '../../modules/create-index/use-partial-filter-expression';
+import { toggleUseTtl } from '../../modules/create-index/use-ttl';
+import { ttlChanged } from '../../modules/create-index/ttl';
+import { toggleUseWildcardProjection } from '../../modules/create-index/use-wildcard-projection';
+import { toggleUseColumnstoreProjection } from '../../modules/create-index/use-columnstore-projection';
+import { wildcardProjectionChanged } from '../../modules/create-index/wildcard-projection';
+import { columnstoreProjectionChanged } from '../../modules/create-index/columnstore-projection';
+import { partialFilterExpressionChanged } from '../../modules/create-index/partial-filter-expression';
+import { toggleUseCustomCollation } from '../../modules/create-index/use-custom-collation';
+import { collationStringChanged } from '../../modules/create-index/collation-string';
+import { openLink } from '../../modules/link';
+import { createIndex } from '../../modules/create-index';
+import { resetForm } from '../../modules/reset-form';
+import CreateIndexForm from '../create-index-form';
+import { toggleUseIndexName } from '../../modules/create-index/use-index-name';
+import type { RootState } from '../../modules/create-index';
+
+const { track } = createLoggerAndTelemetry('COMPASS-IMPORT-EXPORT-UI');
+
+const modalStyles = css({
+  'div[role=dialog] > :first-child': {
+    minHeight: '50vh',
+    maxHeight: '80vh',
+    overflow: 'scroll',
+  },
+});
+
+const modalContentWrapperStyles = css({
+  padding: 'initial',
+});
+
+const modalContentStyles = css({
+  padding: spacing[5],
+});
+
+const bannerStyles = css({
+  marginTop: spacing[3],
+});
+
+const createIndexButtonStyles = css({
+  marginLeft: spacing[2],
+});
+
+/**
+ * Create index modal.
+ */
+function CreateIndexModal({
+  toggleIsVisible,
+  resetForm,
+  isVisible,
+  namespace,
+  error,
+  clearError,
+  inProgress,
+  createIndex,
+  ...props
+}: React.ComponentProps<typeof CreateIndexForm> & {
+  toggleIsVisible: (isVisible: boolean) => void;
+  resetForm: () => void;
+  isVisible: boolean;
+  namespace: string;
+  error?: string;
+  clearError: () => void;
+  inProgress: boolean;
+  createIndex: () => void;
+}) {
+  const onSetOpen = useCallback(
+    (open) => {
+      if (!open) {
+        toggleIsVisible(false);
+        resetForm();
+      } else {
+        track('Screen', { name: 'create_index_modal' });
+      }
+    },
+    [toggleIsVisible, resetForm]
+  );
+
+  const renderError = () => {
+    if (!error) {
+      return;
+    }
+
+    return (
+      <Banner
+        className={bannerStyles}
+        variant="danger"
+        dismissible
+        onClose={clearError}
+      >
+        {error}
+      </Banner>
+    );
+  };
+
+  const renderInProgress = () => {
+    if (!error || !inProgress) {
+      return;
+    }
+
+    return (
+      <Banner className={bannerStyles} variant="info" dismissible>
```

Comment:
```suggestion
      <Banner className={bannerStyles} variant="info">
```
I think we probably don't want this to be `dismissible` if we don't supply an `onClose`.